plugin.program.AML/resources/utils_kodi.py

```diff
@@ -0,0 +1,220 @@
+    dialog = xbmcgui.Dialog()
+    dialog.ok(title, row1, row2, row3)
+
+#
+# Returns True is YES was pressed, returns False if NO was pressed or dialog canceled.
+def kodi_dialog_yesno(row1, row2='', row3='', title = 'Advanced MAME Launcher'):
+    dialog = xbmcgui.Dialog()
+    ret = dialog.yesno(title, row1, row2, row3)
+
+    return ret
+
+#
+# Displays a small box in the bottom right corner
+#
+def kodi_notify(text, title = 'Advanced MAME Launcher', time = 5000):
+    dialog = xbmcgui.Dialog()
+    dialog.notification(title, text, xbmcgui.NOTIFICATION_INFO, time)
+
+def kodi_notify_warn(text, title = 'Advanced MAME Launcher warning', time = 7000):
+    dialog = xbmcgui.Dialog()
+    dialog.notification(title, text, xbmcgui.NOTIFICATION_WARNING, time)
+
+#
+# Do not use this function much because it is the same icon as when Python fails, and that may confuse the user.
+#
+def kodi_notify_error(text, title = 'Advanced MAME Launcher error', time = 7000):
+    dialog = xbmcgui.Dialog()
+    dialog.notification(title, text, xbmcgui.NOTIFICATION_ERROR, time)
+
+def kodi_busydialog_ON():
+    xbmc.executebuiltin('ActivateWindow(busydialog)')
+
+def kodi_busydialog_OFF():
+    xbmc.executebuiltin('Dialog.Close(busydialog)')
+
+def kodi_refresh_container():
+    log_debug('kodi_refresh_container()')
+    xbmc.executebuiltin('Container.Refresh')
+
+# -----------------------------------------------------------------------------
+# Kodi specific stuff
+# -----------------------------------------------------------------------------
+# About Kodi image cache
+#
+# See http://kodi.wiki/view/Caches_explained
+# See http://kodi.wiki/view/Artwork
+# See http://kodi.wiki/view/HOW-TO:Reduce_disk_space_usage
+# See http://forum.kodi.tv/showthread.php?tid=139568 (What are .tbn files for?)
+#
+# Whenever Kodi downloads images from the internet, or even loads local images saved along 
+# side your media, it caches these images inside of ~/.kodi/userdata/Thumbnails/. By default, 
+# large images are scaled down to the default values shown below, but they can be sized 
+# even smaller to save additional space.
+
+#
+# Gets where in Kodi image cache an image is located. 
+#
+def kodi_get_cached_image(image_path):
+    THUMBS_CACHE_PATH = os.path.join(xbmc.translatePath('special://profile/' ), 'Thumbnails')
+
+    # --- Get the Kodi cached image ---
+    # This function return the cache file base name
+    base_name = xbmc.getCacheThumbName(image_path)
+    cache_file_path = os.path.join(THUMBS_CACHE_PATH, base_name[0], base_name)
+
+    return cache_file_path
+
+#
+# Updates Kodi image cache for the image provided with the image itself.
+# In other words, copies the image into Kodi cache entry for the image itself.
+#
+# Needles to say, only update image cache if image already was on the cache.
+def kodi_update_image_cache(img_path):
+    # What if image is not cached?
+    cached_thumb = kodi_get_cached_image(img_path)
```

Comment:
I don't think this is allowed...

Reply:
Please see this comment https://github.com/xbmc/repo-plugins/pull/1886#discussion_r196404466